### PR TITLE
MAT-331: Patch fee data for historic non-default fee donations

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

**Comment by bdsl** [comment]
Getting a test error half the time, not sure why. Just re-running makes it pass which is not great:

```
barney@barney-l-xps:~/projects/matchbot$ docker-compose exec app composer run integration-test
> XDEBUG_MODE=coverage phpunit --config=phpunit-integration.xml --order-by=random
PHPUnit 9.6.10 by Sebastian Bergmann and contributors.

Random Seed:   1699984224

E...............                                                  16 / 16 (100%)

Time: 00:02.876, Memory: 30.00 MB

There was 1 error:

1) MatchBot\IntegrationTests\PatchFeeTest::testItCorrectsFeeRecordOnOldDonation
Error: Call to a member function toArray() on null

/var/www/html/src/Application/Commands/PatchHistoricNonDefaultFeeDonations.php:47
/var/www/html/src/Application/Commands/Command.php:18
/var/www/html/vendor/symfony/console/Command/Command.php:326
/var/www/html/vendor/symfony/console/Tester/CommandTester.php:74
/var/www/html/integrationTests/PatchFeeTest.php:61

ERRORS!
Tests: 16, Assertions: 77, Errors: 1.

Generating code coverage report in Clover XML format ... done [00:00.690]
Script XDEBUG_MODE=coverage phpunit --config=phpunit-integration.xml --order-by=random handling the integration-test event returned with error code 2
barney@barney-l-xps:~/projects/matchbot$ docker-compose exec app composer run integration-test
> XDEBUG_MODE=coverage phpunit --config=phpunit-integration.xml --order-by=random
PHPUnit 9.6.10 by Sebastian Bergmann and contributors.

Random Seed:   1699984240

................                                                  16 / 16 (100%)

Time: 00:02.989, Memory: 28.00 MB

OK (16 tests, 80 assertions)

Generating code coverage report in Clover XML format ... done [00:00.657]
```

File: app/dependencies.php [hunk 1]
```diff
@@ -391,5 +392,19 @@ function (ContainerInterface $c): Auth\DonationPublicAuthMiddleware {
                 new PhpSerializer(),
             );
         },
+        Connection::class => static function (ContainerInterface $c): Connection {
+            $em = $c->get(EntityManagerInterface::class);
+            \assert($em instanceof EntityManagerInterface);
```
Comment (bdsl) [inline app/dependencies.php:397]: getting quite bored of all these asserts, think I'll put a Psalm stub for the ContainerInterface in next time that says that it always `T` when we pass `class-string<T>`

**Review by NoelLH — COMMENTED** [review]
Looks good from a v quick review, not sure about the intermittent test from first glance

**Comment by bdsl** [comment]
@bdsl 

> Getting a test error half the time, not sure why. Just re-running makes it pass which is not great:

53b3f0eee7020e8 should fix it.

**Review by dorota-joanna — APPROVED** [review]
LGTM

**Pull request merged**